src/lib/tuning/abstract_selector.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "tuning/tuning_choice.hpp"
+#include "tuning/tuning_operation.hpp"
+
+namespace opossum {
+
+/**
+ * An AbstractSelector transforms an unordered list of TuningChoices into
+ * a concrete sequence of TuningOperations.
+ *
+ * It considers a cost budget that the entire operation sequence as well as any
+ * continuous subsequence from the beginning must not exceed.
+ * The operation sequence is prioritized by expected performance impact on the system,
```

Comment:
Could you expand on what you mean by „performance impact“? This could be anything: Cost of choosing that option, memory usage, ...

Reply:
True, this was misleading. Actually we meant positive performance impact, so I changed it to *performance improvement*.

Reply:
My thought was that the word "performance" is too vague.